### PR TITLE
Feature/update student state

### PR DESCRIPTION
**Pull request by Moussa348**
*update student state
*adding validating page request to FileMetaDataService --> getListInvalidCv and returning status on invalid request

### CODE REVIEW

File: backend/src/main/java/com/team4/backend/controller/StudentController.java [hunk 1]
```diff
@@ -30,4 +34,15 @@ public StudentController(StudentService studentService) {
         //TODO add a non-handled exception to make sure it returns 500 and not 409
     }
 
+    @PatchMapping("/updateStudentState")
+    @PreAuthorize("hasAuthority('STUDENT')")
+    public Mono<ResponseEntity<String>> updateStudentState(Principal principal) {
+        return studentService.updateStudentState(UserSessionService.getLoggedUserEmail(principal), StudentState.INTERNSHIP_FOUND)
+                .flatMap(s -> Mono.just(ResponseEntity.status(HttpStatus.NO_CONTENT).body("")))
```
Comment (0xD34DC0DE) [inline backend/src/main/java/com/team4/backend/controller/StudentController.java:41]: you can filter exceptions with onErrorResume by putting the ExceptionName.class as the first argument.
So you could have 2 onErrorResume instead

File: backend/src/main/java/com/team4/backend/exception/InvalidPageRequestException.java [hunk 2]
```diff
@@ -1,5 +1,8 @@
 package com.team4.backend.exception;
 
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
```
Comment (0xD34DC0DE) [inline backend/src/main/java/com/team4/backend/exception/InvalidPageRequestException.java:5]: I see the ResponseStatus import but no annotations and @NoArgsConstructor is missing

File: backend/src/main/java/com/team4/backend/exception/ForbiddenActionException.java [hunk 3]
```diff
@@ -0,0 +1,11 @@
+package com.team4.backend.exception;
+
+import lombok.NoArgsConstructor;
+
+@NoArgsConstructor
```
Comment (0xD34DC0DE) [inline backend/src/main/java/com/team4/backend/exception/ForbiddenActionException.java:5]: Missing the @ResponseStatus annotation

File: backend/src/main/java/com/team4/backend/model/enums/StudentState.java [hunk 4]
```diff
@@ -1,5 +1,5 @@
 package com.team4.backend.model.enums;
 
 public enum StudentState {
-    REGISTERED
+    REGISTERED,INTERNSHIP_NOT_FOUND, INTERNSHIP_FOUND,WAITING_FOR_RESPONSE
```
Comment (0xD34DC0DE) [inline backend/src/main/java/com/team4/backend/model/enums/StudentState.java:4]: Formatting

File: backend/src/main/java/com/team4/backend/service/StudentService.java [hunk 5]
```diff
@@ -50,8 +53,19 @@ public StudentService(StudentRepository studentRepository, PBKDF2Encoder pbkdf2E
...
+                .map(student -> {
+                    //TODO --> call function that will trigger the contract generation
+                    student.setStudentState(studentState);
+                    return student;
+                }).flatMap(studentRepository::save);
+    }
+
     public Mono<Student> addOfferToStudentAppliedOffers(Student student, String offerId) {
-        if(!student.getAppliedOffersId().contains(offerId)) {
+        if (!student.getAppliedOffersId().contains(offerId)) {
```
Comment (0xD34DC0DE) [inline backend/src/main/java/com/team4/backend/service/StudentService.java:68]: formatting

File: backend/src/main/java/com/team4/backend/controller/FileMetaDataController.java [hunk 6]
```diff
@@ -40,8 +41,9 @@
 
     @GetMapping("/getListInvalidCvNotSeen/{noPage}")
     @PreAuthorize("hasAuthority('INTERNSHIP_MANAGER')")
-    public Flux<FileMetaDataInternshipManagerViewDto> getListInvalidCvNotSeen(@PathVariable Integer noPage) throws InvalidPageRequestException {
-        return fileMetaDataService.getListInvalidCvNotSeen(noPage).map(FileMetaDataMapper::toInternshipManagerViewDto);
+    public Flux<FileMetaDataInternshipManagerViewDto> getListInvalidCvNotSeen(@PathVariable Integer noPage) {
+        return fileMetaDataService.getListInvalidCvNotSeen(noPage).map(FileMetaDataMapper::toInternshipManagerViewDto)
+                .onErrorMap(error -> new ResponseStatusException(HttpStatus.BAD_REQUEST,error.getMessage()));
```
Comment (0xD34DC0DE) [inline backend/src/main/java/com/team4/backend/controller/FileMetaDataController.java:46]: What can cause a malformed request ?

**Review by 0xD34DC0DE — CHANGES_REQUESTED** [review]

File: backend/src/main/java/com/team4/backend/exception/ForbiddenActionException.java (hunk 3 above)
Comment (Moussa348) [reply backend/src/main/java/com/team4/backend/exception/ForbiddenActionException.java:5]: I'm following our convention, we return a responseStatusException already so no need, we will only repeat ourself

File: backend/src/main/java/com/team4/backend/controller/FileMetaDataController.java (hunk 6 above)
Comment (Moussa348) [reply backend/src/main/java/com/team4/backend/controller/FileMetaDataController.java:46]: when it is negative or null

File: backend/src/main/java/com/team4/backend/exception/ForbiddenActionException.java (hunk 3 above)
Comment (0xD34DC0DE) [reply backend/src/main/java/com/team4/backend/exception/ForbiddenActionException.java:5]: If we follow our conventions, the service should throw the exception

**Pull request merged**